compiler/rustc_resolve/src/imports.rs

```diff
@@ -1168,8 +1168,15 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     target_bindings[ns].get(),
                 ) {
                     Ok(other_binding) => {
+                        let key = this.new_key(target, ns);
                         is_redundant[ns] = Some(
-                            binding.res() == other_binding.res() && !other_binding.is_ambiguity(),
+                            binding.res() == other_binding.res()
```

Comment:
Why does this condition hold for #92904 if the `Foo`s imported by ` use bar::*` and `use self::Foo` are clearly different?

Reply:
This lint is *supposed* to report single imports that can be removed because a glob imports the same thing.
The `!other_binding.is_ambiguity()` condition makes sure that single imports are not removed in situations where they are used for disambiguation.

#92904, however, is not about disambiguation, so there's some other bug responsible for reporting the wrong warning here (maybe in accumulation over 3 namespaces or something).